# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.h
# Commit: f505b8fb8f8b
# Full Hash: f505b8fb8f8b7a8784be192ce6a36dc2278095cb
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 8: Use TaggedParserAtomIndex in ScriptStencil.functionAtom. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96915
# ==============================================================================

diff -r 3749d6d2325d -r f505b8fb8f8b js/src/frontend/Stencil.h
--- a/js/src/frontend/Stencil.h	Sat Nov 14 00:03:00 2020 +0000
+++ b/js/src/frontend/Stencil.h	Sat Nov 14 00:03:07 2020 +0000
@@ -18,6 +18,7 @@
 #include "frontend/AbstractScopePtr.h"    // AbstractScopePtr, ScopeIndex
 #include "frontend/FunctionSyntaxKind.h"  // FunctionSyntaxKind
 #include "frontend/ObjLiteral.h"          // ObjLiteralStencil
+#include "frontend/ParserAtom.h"          // TaggedParserAtomIndex
 #include "frontend/TypedIndex.h"          // TypedIndex
 #include "js/RegExpFlags.h"               // JS::RegExpFlags
 #include "js/RootingAPI.h"                // Handle
@@ -474,7 +475,7 @@
 
   // The explicit or implicit name of the function. The FunctionFlags indicate
   // the kind of name.
-  const ParserAtom* functionAtom = nullptr;
+  TaggedParserAtomIndex functionAtom;
 
   // See: `FunctionFlags`.
   FunctionFlags functionFlags = {};
@@ -532,8 +533,8 @@
 
 #if defined(DEBUG) || defined(JS_JITSPEW)
   void dump();
-  void dump(JSONPrinter& json);
-  void dumpFields(JSONPrinter& json);
+  void dump(JSONPrinter& json, CompilationStencil* compilationStencil);
+  void dumpFields(JSONPrinter& json, CompilationStencil* compilationStencil);
 #endif
 };
 